### PR TITLE
DSND-2372: Transform top level annotation to api request

### PR DESCRIPTION
**Pull request by sthompsonCH**
### Related Jira tickets
[DSND-2372](https://companieshouse.atlassian.net/browse/DSND-2372)

### General
- [ ] Is the PR as small as it can be?
- [ ] Has the Jira ticket been updated with any relevant comments?
- [ ] Is the `README` up to date?
- [ ] Has the `POM` been updated for the latest versions of dependencies?
- [ ] Has the code been double-checked against `main`?
- [ ] Does the code adhere standards in this repository, including SonarQube checks?

### Testing
- [ ] Do the code changes have unit and integration tests with code coverage > 80%?
- [ ] Has the code been tested locally and/or passed the API Karate tests on Tilt?
- [ ] Have mandatory manual test cases been run?
- [ ] Are extra Karate tests required?
- [ ] Are all the test data resources up to date?

### Release preparation
_Where possible, add links to the respective Jira tickets when an item is checked_
- [ ] Have these changes been included in a release ticket?
- [ ] Are changes required to the `docker-chs-development` deployment or test data?
- [ ] Are any changes required to the environment added to `chs-configs`?

## Notes to the tester
_Add any testing hints or instructions here._

[DSND-2372]: https://companieshouse.atlassian.net/browse/DSND-2372?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -31,7 +31,7 @@
     <!--companies house-->
     <structured-logging.version>3.0.1</structured-logging.version>
     <sdk-manager-java.version>3.0.3</sdk-manager-java.version>
-    <private-api-sdk-java.version>4.0.106</private-api-sdk-java.version>
+    <private-api-sdk-java.version>unversioned</private-api-sdk-java.version>
```
Comment (johncook-ch) [inline pom.xml:34]: Need a proper version

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/filinghistory/consumer/mapper/posttransform/InternalFilingHistoryApiMapper.java [hunk 2]
```diff
@@ -41,11 +41,13 @@ public InternalFilingHistoryApi mapInternalFilingHistoryApi(InternalFilingHistor
 
         String barcode = getFieldValueFromJsonNode(topLevelNode, "_barcode");
         String documentId = getFieldValueFromJsonNode(topLevelNode, "_document_id");
+        String matchedDefault = getFieldValueFromJsonNode(topLevelNode, "matched_default");
 
         InternalData internalData = new InternalData()
                 .originalDescription(getFieldValueFromJsonNode(topLevelNode, "original_description"))
                 .parentEntityId(getFieldValueFromJsonNode(topLevelNode, "parent_entity_id"))
                 .entityId(getFieldValueFromJsonNode(topLevelNode, "_entity_id"))
+                .matchedDefault(matchedDefault == null ? null : Integer.parseInt(matchedDefault))
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/filinghistory/consumer/mapper/posttransform/InternalFilingHistoryApiMapper.java:50]: Need to add test coverage for this condition

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by johncook-ch — APPROVED** [review]

**Review by kinpang-CH — APPROVED** [review]

**Pull request merged**